ethicml/utility/data_structures.py

```diff
@@ -328,18 +345,31 @@ def concat_tt(
     )
 
 
-FairnessType: TypeAlias = Literal["DP", "EqOp", "EqOd"]  # pylint: disable=invalid-name
+@enum_name_str
+class FairnessType(Enum):
+    """Fairness type."""
+
+    dp = auto()
+    eqop = auto()
+    eqod = auto()
```

Comment:
are you preferring lower case now? (not that I mind)

Reply:
It's not so much that I prefer lowercase, but trying to make the de enums consistent. Right now we have `LR`, `linear` and `EqOp`. So I'm making everything lowercase now and can review this decision later.

Reply:
I thought that there was a PEP saying enums should be upper case, but I just looked it up. The PEP for enums has lowercase, but Guido later decreed that enums should be upper case as they're constants

Reply:
Looking at other languages:

- in swift enum values have the same capitalization as normal variables: https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html
- in rust, they're capitalized like classes: https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html
- in kotlin, they're all caps: https://kotlinlang.org/docs/enum-classes.html
- in typescript, like rust, with capitalized CamelCase: https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
across our other projects, @MylesBartlett has decided to go with lowercase enums. I think this decision was made to make setting values with hydra look nice. I think it's fine to go with all lowercase, but I could be persuaded on all uppercase without much effort. I'm less convinced on CamelCase, but again, it wouldn't take _that_ much to persuade me.